### PR TITLE
flatcam: init at 8.5

### PR DESCRIPTION
**Pull request by trepetti**
###### Motivation for this change

[FlatCAM](http://flatcam.org) is a software for 2D design postprocessing for manufacturing focusing on fabricating PCB designs on CNC routers. The software lets users take Gerber and Excellon files (industry standard PCB layer information files) and converts them to gcode to be executed on a CNC. Nix already has many PCB and CNC packages available, and this one fills in a gap. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by symphorien — COMMENTED** [review]
Overall, this looks good, thanks!
I left a few comments, but they are no blocker.

File: pkgs/applications/science/electronics/flatcam/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "flatcam";
+  version = "8.5";
+
+  src = fetchFromBitbucket {
+    owner = "jpcgt";
+    repo = "${pname}";
+    rev = "533afd6a1772857cb633c011b5e0a15b60b1e92e"; # 8.5 with Red Hat packaging.
```
[the remark this answers is not in the document]
Comment (trepetti) [reply pkgs/applications/science/electronics/flatcam/default.nix:14]: They do in fact have tags, but the commit that switched to conventional `setup.py` + `requirements.txt` packaging was done shortly after the last major release and a change in maintainers. The contributor of this commit is the maintainer for the RPM package in Fedora, so sadly this didn't make it into a major release version. There is a ton of work going on right now on a new development branch that will ultimately result in 9.0, but things are a bit messy at the moment. I preemptively packaged `svglib` the last major dependency missing for what will end up being 9.0, and am planning on submitting some pull requests upstream shortly to make building wheels easier and integrate into the Nix flow better. I decided against packaing the beta branch for now.

[the remark this answers is not in the document]
Comment (trepetti) [reply pkgs/applications/science/electronics/flatcam/default.nix:14]: I can submit a PR to the stable branch, though, too. And maybe request an 8.5.1 tag.

File: pkgs/applications/science/electronics/flatcam/default.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    scipy
+    setuptools
+    shapely
+    simplejson
+    six
+    svg-path
+  ];
+
+  patches = [
+    ./fix_setup_manifest_version_and_missing_resources.patch
```
[the remark this answers is not in the document]
Comment (trepetti) [reply pkgs/applications/science/electronics/flatcam/default.nix:36]: Yup, will do. Will submit a PR upstream to fix these issues a long with the request for an 8.5.1 tag.

**Comment by trepetti** [comment]
I have a pending PR with those fixes up-stream. That beta branch I mentioned too may not be merged anytime soon but adds some cool features (and also drops Qt4, which I know Nix will be doing soon too). I will package it separately as flatcam-beta.

**Review by symphorien — APPROVED** [review]

**Pull request merged**